plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
At most I'd put a db.Ping() at the start it will cause it to establish a working connection.